extensions/amp-form/0.1/amp-form.js

```diff
@@ -216,6 +221,29 @@ export class AmpForm {
   }
 
   /**
+   * Returns form data as an object.
+   * @return {!Object}
+   * @private
+   */
+  getFormAsObject_() {
+    const data = {};
+    const inputs = this.form_.querySelectorAll('input,select,textarea');
+    for (let i = 0; i < inputs.length; i++) {
+      const input = inputs[i];
+      if (!input.name || !input.value) {
```

Comment:
I also don't think we should exclude inputs without `value`. That would be different from `post`. Presence of the key might be expected on the server side.